### PR TITLE
[WIP] add readme when .pytest_cache is created

### PR DESCRIPTION
**Pull request by avirlrma**
WIP PR for #3519.

### CODE REVIEW

**Comment by avirlrma** [comment]
Hi @nicoddemus 
I have made a PR with changes as discussed. I still have to add tests for this and work on the Environment variable that you mentioned in the ticket description

File: src/_pytest/cacheprovider.py [hunk 1]
```diff
@@ -85,6 +85,13 @@ def set(self, key, value):
                python types, including nested types
                like e. g. lists of dictionaries.
         """
+        content_readme = 'test'
```
Comment (nicoddemus) [inline src/_pytest/cacheprovider.py]: I suggest to move this block of code to its own function to improve readability, something like `_ensure_readme()` perhaps?

File: src/_pytest/cacheprovider.py [hunk 2]
```diff
@@ -85,6 +85,13 @@ def set(self, key, value):
...
+        if os.path.isdir(self._cachedir):
+            readme_path = os.path.join(self._cachedir, 'README.md')
+            if os.path.isfile(readme_path) is False:
```
Comment (nicoddemus) [inline src/_pytest/cacheprovider.py]: The idiomatic way is to use `if not os.path.isfile(readme_path)`. Also, it is probably easier to use the [`py.path` API](http://py.readthedocs.io/en/latest/path.html) directly:

```python
        if self._cachedir.check(dir=True):
            readme_path = self._cachedir.join('README.md')
            if not readme_path.check(file=True):
                readme_path.write(content_readme)
```

File: src/_pytest/cacheprovider.py (hunk 1 above)
Comment (avirlrma) [reply src/_pytest/cacheprovider.py]: oh yes! That was so evident. Stupid me. Sorry.

**Comment by avirlrma** [comment]
@nicoddemus 
I made the changes as suggested by you. Sorry for overlooking the `black` part in guidelines.
Also I moved the call to `_ensure_readme()` to the bottom of `set()`,so that the permissions `try and except` block can do their work and the code won't be repeated.

I have updated the above checklist.Please check if it's all that neccesary and enough.

File: src/_pytest/cacheprovider.py [hunk 3]
```diff
@@ -103,6 +103,14 @@ def set(self, key, value):
             with f:
                 self.trace("cache-write %s: %r" % (key, value))
                 json.dump(value, f, indent=2, sort_keys=True)
+        self._ensure_readme()
```
Comment (nicoddemus) [inline src/_pytest/cacheprovider.py]: This probably should be indented with the `with f` statement above; this way we ensure we have written the key/value and will be able to write the readme file as well.

**Comment by nicoddemus** [comment]
>Sorry for overlooking the black part in guidelines.

No need to be sorry at all, that's what PRs are for! 👍

File: src/_pytest/cacheprovider.py [hunk 4]
```diff
@@ -103,6 +103,14 @@ def set(self, key, value):
...
+
+    def _ensure_readme(self):
+        content_readme = "test"
```
Comment (nicoddemus) [inline src/_pytest/cacheprovider.py]: We need to include some actual contents. How about:

```markdown
# pytest cache directory #

This directory contains data from the pytest's cache plugin, which provides the `--lf` and `--ff` options, as well as the `cache` fixture. 

**Do not** commit this to version control.

See [the docs](https://docs.pytest.org/en/latest/cache.html) for more information.
```

Comment (avirlrma) [reply src/_pytest/cacheprovider.py]: looks good.I will add it in

**Comment by RonnyPfannschmidt** [comment]
please rebase, those merges are a massive problem

**Comment by avirlrma** [comment]
@RonnyPfannschmidt @nicoddemus I couldn't solve what had happened. Anyways I'm made a new PR #3608,sorry for the trouble caused.

**Pull request closed without merging**